eager quant: convert mapping for fused QAT Linear-Bn1d

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72796
* #72431

Summary:

Adds the eager mode convert mapping for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34213150](https://our.internmc.facebook.com/intern/diff/D34213150)

vkuzo committed:
eager quant: convert mapping for fused QAT Linear-Bn1d

Summary:

Adds the eager mode convert mappint for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).

jerryzh168 commented on test/quantization/eager/test_quantize_eager_qat.py:
```diff
@@ -1006,6 +1006,22 @@ def test_linear_bn_numerics(self):
+        m = nn.Sequential(
+            QuantStub(),
+            nn.Linear(4, 4),
+            nn.BatchNorm1d(4),
```
optional nit: maybe add a DeQuantStub, just for completeness..

jerryzh168 commented:
looks like there are some errors from numeric suite

vkuzo committed:
Update on "eager quant: convert mapping for fused QAT Linear-Bn1d"


Summary:

Adds the eager mode convert mapping for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34213150](https://our.internmc.facebook.com/intern/diff/D34213150)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).

vkuzo committed:
Update on "eager quant: convert mapping for fused QAT Linear-Bn1d"


Summary:

Adds the eager mode convert mapping for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34213150](https://our.internmc.facebook.com/intern/diff/D34213150)

[ghstack-poisoned]

vkuzo committed:
Update on "eager quant: convert mapping for fused QAT Linear-Bn1d"


Summary:

Adds the eager mode convert mapping for fused QAT Linear-Bn1d module.

Test Plan:

```
python test/test_quantization.py TestQuantizeEagerQATNumerics.test_linear_bn_workflow
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34213150](https://our.internmc.facebook.com/intern/diff/D34213150)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34213150).
